### DISCUSSION TITLE
how to run alpine in qemu with multi-core

### DISCUSSION THREAD

**Discussion by Angel0726** [termux/termux-packages#8755] [Q&A] [answered]
I found that alpine in qemu is too slow; what's i use `top` in termux, i find only one core works. so how to run alpine in qemu with multi-core. I test `qemu-system-x86_64 -hda virt-alpine.img -boot c -m 1024 -smp 6,sockets=6`, but it does not work
![image](https://user-images.githubusercontent.com/32816780/150999190-d0a2c218-5bba-471d-8a66-1bd9a1558a64.png)

**Comment by ghost** [comment, marked as answer]
It will not work. You can emulate multiple CPU cores, but they all will be executed within **one** host QEMU thread. Also multiple CPUs work only for multithreaded programs. Single-threaded programs consume exactly one CPU.

You need MTTCG setup, but this one has issues when host architecture doesn't match the target.

**Reply by Angel0726** [reply]
thank you
